web/src/test/java/com/replica/RouterServerTest.java

```diff
@@ -458,12 +509,28 @@ public void testAutoQueryCustomSpeeds() {
             RouterOuterClass.StreetPath customSpeedsPath = Iterables.getOnlyElement(customSpeedsResponse.getPathsList());
             RouterOuterClass.StreetPath defaultSpeedsPath = Iterables.getOnlyElement(defaultSpeedsResponse.getPathsList());
 
+            double customSpeed = Double.parseDouble(getOnlyPathDetailValue(ReplicaPathDetails.SPEED, customSpeedsPath));
+            assertEquals(THURTON_DRIVE_CUSTOM_SPEED, customSpeed);
+
+            double defaultSpeed = Double.parseDouble(getOnlyPathDetailValue(ReplicaPathDetails.SPEED, defaultSpeedsPath));
             // the custom speeds profile sets the Thurton Drive speed very high, so the travel time using this profile
             // should be less than the default
+            assertTrue(defaultSpeed < customSpeed);
             assertTrue(customSpeedsPath.getDurationMillis() < defaultSpeedsPath.getDurationMillis());
+
+            // both paths should only involve the Thurton drive OSM way
+            for (RouterOuterClass.StreetPath path : Arrays.asList(customSpeedsPath, defaultSpeedsPath)) {
+                int osmId = Integer.parseInt(getOnlyPathDetailValue(ReplicaPathDetails.OSM_ID, path));
+                assertEquals(THURTON_DRIVE_OSM_ID, osmId);
+            }
         }
     }
 
+    private static String getOnlyPathDetailValue(String detailName, RouterOuterClass.StreetPath streetPath) {
+        Map<String, List<RouterOuterClass.StreetPathDetailValue>> customPathDetailsByName = getPathDetailsByName(streetPath);
+        return Iterables.getOnlyElement(customPathDetailsByName.get(detailName)).getValue();
```

Comment:
Is the assumption here that the speed along all links in each path we pass to `getOnlyPathDetailValue` should be the same? ie there's a single `StreetPathDetailValue` for speed in each path?

That makes sense if the O/D we use gives us routes with this property, just wanted to clarify for my own understanding

Reply:
Ah I see the later comment `both paths should only involve the Thurton drive OSM way` for checking a similar property w/ the OSM ID - seems like ^ is the case

Reply:
Yep, exactly! Added a code comment to help clarify